lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -13,6 +13,65 @@ use std::slice;
 use std::slice::from_raw_parts;
 use std::sync::Arc;
 
+use raw_cpuid::CpuId;
+
+fn check_feature_support(module_features: &ModuleFeatures) -> Result<(), Error> {
+    let cpuid = CpuId::new();
+
+    fn missing_feature(feature: &str) -> Error {
+        Error::Unsupported(format!(
+            "Module requires feature host does not support: {}",
+            feature
+        ))
+    }
+
+    let info = cpuid
+        .get_feature_info()
+        .ok_or_else(|| Error::Unsupported("Unable to obtain host CPU feature info!".to_string()))?;
```

Comment:
I'm not sure how expensive this is, but we could put it in a `std::sync::Once` if it moves the needle.